BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Models/SpanModel.cs

```diff
@@ -3,71 +3,116 @@
 
 namespace BugsnagUnityPerformance
 {
-   internal class SpanModel
-{
-    static readonly DateTimeOffset _unixStart = new DateTimeOffset(1970, 1, 1, 0, 0, 0, TimeSpan.Zero);
-
-    public string name;
-    public int kind;
-    public string spanId;
-    public string traceId;
-    public string startTimeUnixNano;
-    public string endTimeUnixNano;
-    public string parentSpanId;
-    public List<AttributeModel> attributes = new List<AttributeModel>();
-
-    public SpanModel(Span span)
+    internal class SpanModel
     {
-        name = span.Name;
-        kind = (int)span.Kind;
-        spanId = span.SpanId;
-        traceId = span.TraceId.Replace("-", string.Empty);
-        parentSpanId = span.ParentSpanId;
-        startTimeUnixNano = GetNanoSeconds(span.StartTime);
-        endTimeUnixNano = GetNanoSeconds(span.EndTime);
+        static readonly DateTimeOffset _unixStart = new DateTimeOffset(1970, 1, 1, 0, 0, 0, TimeSpan.Zero);
+        public string name;
+        public int kind;
+        public string spanId;
+        public string traceId;
+        public string startTimeUnixNano;
+        public string endTimeUnixNano;
+        public string parentSpanId;
+        public List<AttributeModel> attributes = new List<AttributeModel>();
 
-        foreach (var attr in span.GetAttributes())
+        public SpanModel(Span span, int attributeArrayLengthLimit, int attributeStringValueLimit)
         {
-            if (attr.Value is string[] stringArray)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeStringArrayValueModel(stringArray)));
-            }
-            else if (attr.Value is long[] intArray)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeIntArrayValueModel(intArray)));
-            }
-            else if (attr.Value is bool[] boolArray)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeBoolArrayValueModel(boolArray)));
-            }
-            else if (attr.Value is double[] doubleArray)
-            {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeDoubleArrayValueModel(doubleArray)));
-            }
-            else if (attr.Value is string strValue)
+            name = span.Name;
+            kind = (int)span.Kind;
+            spanId = span.SpanId;
+            traceId = span.TraceId.Replace("-", string.Empty);
+            parentSpanId = span.ParentSpanId;
+            startTimeUnixNano = GetNanoSeconds(span.StartTime);
+            endTimeUnixNano = GetNanoSeconds(span.EndTime);
+
+            foreach (var attr in span.GetAttributes())
             {
-                attributes.Add(new AttributeModel(attr.Key, new AttributeStringValueModel(strValue)));
+                if (string.IsNullOrEmpty(attr.Key) || attr.Value == null)
+                {
+                    continue;
+                }
+
+                if (attr.Key.Length > 128)
```

Comment:
Suggest abstracting this to a constant, just to make it easier to find/understand as a fixed limit.

e.g. `MAXIMUM_ATTRIBUTE_KEY_LENGTH_LIMIT`